engine/table/src/main/java/io/deephaven/engine/table/impl/lang/QueryLanguageParser.java

```diff
@@ -944,6 +944,27 @@ public static boolean isTypedVector(Class<?> type) {
     private Expression[] convertParameters(final Executable executable,
             final Class<?>[] argumentTypes, final Class<?>[] expressionTypes,
             final Class<?>[][] parameterizedTypes, Expression[] expressions) {
+
+        // For Python callables, all Vector columns should be converted into arrays
+        if (executable.getDeclaringClass() == PyCallableWrapper.class) {
+            for (int ei = 0; ei < expressionTypes.length; ei++) {
+                if (isTypedVector(expressionTypes[ei])) {
+                    expressions[ei] =
+                            new MethodCallExpr(new NameExpr("VectorConversions"), "nullSafeVectorToArray",
+                                    new NodeList<>(expressions[ei]));
+                    // To prevent JPY from unpacking a non-primitive array to make sure the result array is treated
+                    // as a single argument
+                    if (expressionTypes[ei] == ObjectVector.class) {
+                        expressions[ei] = new CastExpr(
+                                new ClassOrInterfaceType("java.lang.Object"),
+                                expressions[ei]);
+                    }
+                    expressionTypes[ei] = convertVector(expressionTypes[ei],
+                            parameterizedTypes[ei] == null ? null : parameterizedTypes[ei][0]);
```

Comment:
I'm nervous that in the case of `ObjectVector` we are creating an expression that explicitly conflicts with the expression type. Maybe we need the code to be this:
```suggestion
                    if (expressionTypes[ei] == ObjectVector.class) {
                        expressions[ei] = new CastExpr(
                                new ClassOrInterfaceType("java.lang.Object"),
                                expressions[ei]);
                        expressionTypes[ei] = Object.class;
                    } else {
                        expressionTypes[ei] = convertVector(expressionTypes[ei],
                                parameterizedTypes[ei] == null ? null : parameterizedTypes[ei][0]);
                    }
```